spring-batch-core/src/main/java/org/springframework/batch/core/repository/dao/StepExecutionDao.java

```diff
@@ -86,4 +86,13 @@ default StepExecution getLastStepExecution(JobInstance jobInstance, String stepN
 	 */
 	void addStepExecutions(JobExecution jobExecution);
 
+	/**
+	 * Counts all the {@link StepExecution} for a given step name.
+	 *
+	 * @param jobInstance the parent {@link JobInstance}
+	 * @param stepName the name of the step
+	 * @since 4.2.2
```

Comment:
The plan is to introduce this improvement in the upcoming 4.3. So please update the `since` tag accordingly.

Reply:
Changed since-tag to next minor version